crates/turborepo-cache/src/cache_archive/restore_symlink.rs

```diff
@@ -0,0 +1,150 @@
+use std::{backtrace::Backtrace, path::Path};
+
+use camino::{Utf8Path, Utf8PathBuf};
+use path_clean::clean;
+use turbopath::{AbsoluteSystemPath, AnchoredSystemPath, AnchoredSystemPathBuf, PathError};
+
+use crate::{
+    cache_archive::{restore::canonicalize_name, restore_directory::CachedDirTree},
+    CacheError,
+};
+
+pub fn restore_symlink(
+    dir_cache: &mut CachedDirTree,
+    anchor: &AbsoluteSystemPath,
+    header: &tar::Header,
+) -> Result<AnchoredSystemPathBuf, CacheError> {
+    let processed_name = canonicalize_name(&header.path()?)?;
+
+    let processed_linkname = canonicalize_linkname(
+        anchor,
+        &processed_name,
+        &header.link_name()?.expect("has linkname"),
```

Comment:
Might be good to pull this out once so we don't have `.expect` more than we need to.

Reply:
In fact, rather than crash, we probably ought to report a malformed tar. I suspect it's possible to have a the bit set for a symlink without having a link name present.